### PR TITLE
Adding tests for S3 Result

### PR DESCRIPTION
**Pull request by Nyholm**

### CODE REVIEW

File: src/Flysystem/S3/Tests/Integration/S3FilesystemV2Test.php [hunk 1]
```diff
@@ -108,6 +108,6 @@ private function s3Client(): S3Client
         $this->shouldCleanUp = true;
         $options = ['accessKeyId' => $key, 'accessKeySecret' => $secret, 'region' => $region];
 
-        return $this->s3Client = new S3Client($options, new NullProvider());
```
Comment (jderusse) [inline src/Flysystem/S3/Tests/Integration/S3FilesystemV2Test.php]: if you remove this,  CI tests will call the AWS endpoint to fetch instance Role, that slows tests.
You should either :
- check if class NullProvider exists
-  use random credentials in options parameters

Comment (Nyholm) [reply src/Flysystem/S3/Tests/Integration/S3FilesystemV2Test.php]: That is fine for this one. This is only used when some environment variables are being set. Ie, it is not used on CI only when you really want to test with aws servers.

Comment (jderusse) [reply src/Flysystem/S3/Tests/Integration/S3FilesystemV2Test.php]: are you sure?
by default it:
- look for credentials in options
- look for credentials in HOME/.AWS/credentials
- call AWS endpoint

given in CI we provides neither option nor config file, it always call AWS endpoint.

Comment (Nyholm) [reply src/Flysystem/S3/Tests/Integration/S3FilesystemV2Test.php]: Yes, have a look here: 
https://github.com/async-aws/aws/blob/master/src/Flysystem/S3/Tests/Integration/S3FilesystemV2Test.php#L96-L111

Comment (jderusse) [reply src/Flysystem/S3/Tests/Integration/S3FilesystemV2Test.php]: ahhhh. got it.   sorry

**Pull request merged**